Initial commit

hieunq95 opened the pull request:
Project for review

hieunq95 committed:
Initial commit

quan-dang commented on docker-compose.yml:
```diff
@@ -0,0 +1,65 @@
+        - ./rag-pipeline/vector_store:/rag-pipeline/vector_store
+      networks:
+        - local-net
+      healthcheck:
```
Nếu đã cho health check ở đây thì nên cho cả các service khác nữa nhé sếp, k thì nhìn hơi lạc quẻ với giống copy paste chỗ khác vào.

quan-dang commented on jenkins/Jenkinsfile:
Thường thì file này để ở root path luôn nhé sếp

quan-dang commented on jenkins/Jenkinsfile:
```diff
@@ -0,0 +1,62 @@
+            steps {
+                script {
+                    echo 'Running unit tests on backend'
+                    sh 'docker exec rag-pipeline bash -c "export PYTHONPATH=/rag-pipeline/src && pytest test "'
```
Test trước r mới build chứ sếp nhỉ?

quan-dang commented on jenkins/Jenkinsfile:
```diff
@@ -0,0 +1,62 @@
+        stage('Deploy') {
+            steps {
+                script {
+                    echo 'Deploying services...'
```
Chỗ này nếu đã deploy bằng docker compose rồi thì bỏ đi sếp, cơ mà đồ án của mình là phải deploy trên k8s đó nhé sếp.

quan-dang commented on jenkins/entrypoint.sh:
```diff
@@ -0,0 +1,14 @@
+
+if [ -S /var/run/docker.sock ]; then
+    echo "Fixing permissions for Docker socket..."
+    sudo chmod 666 /var/run/docker.sock
```
Không nên set 666 nhé sếp, sếp có thể đọc thêm tại: https://stackoverflow.com/a/47183372

quan-dang commented on kubernetes/backend-service.yaml:
```diff
@@ -0,0 +1,14 @@
+kind: Service
+metadata:
+  name: rag-pipeline
+  namespace: hieunq95
```
ko ai đặt tên namespace là tên mình nhé sếp, nên đặt theo chức năng, ví dụ model-serving

quan-dang commented on kubernetes/backend-service.yaml:
```diff
@@ -0,0 +1,14 @@
+    - protocol: TCP
+      port: 8000   # Service port
+      targetPort: 8000 # Container port
+      nodePort: 30080  # Exposed port on your machine (must be between 30000-32767)
```
Sếp cẩn thận quá =)) còn ghi luôn range

quan-dang commented on kubernetes/deployment.yaml:
```diff
@@ -0,0 +1,45 @@
+    spec:
+      containers:
+        - name: backend
+          image: hieunq95/tiny-llm-agent-rag-pipeline:latest
```
image k nên để latest mà nên ghi rõ version

quan-dang commented on kubernetes/helm-chart/templates/namespace.yaml:
Thường k nên để namespace trong chart sếp ạ, dễ gây lỗi nếu namespace được tạo sau ứng dụng.

quan-dang commented on rag-pipeline/src/data_extraction.py:
```diff
@@ -0,0 +1,37 @@
+def extract_data(pdf_path: str) -> list:
+    """Extract text from a PDF document and return a list of text chunks.
+
+    Args:
```
Cái hàm này có dùng chatgpt k thế sếp =))

quan-dang commented on rag-pipeline/src/main.py:
```diff
@@ -0,0 +1,177 @@
+        logger.error(f"❌ LLM Model Load Failed: {e}", exc_info=True)
+        LLM_LOADED = False
+        
+def monitor_memory_usage(interval: int=5):
```
nên tách ra 1 file utils rieeng nhé sếp

quan-dang replied:
cơ mà em nghĩ cũng k cần 1 thread rieeng cho cái này =)) vì memory của process đều được log ở OS hết.

quan-dang commented on rag-pipeline/src/model_setup.py:
```diff
@@ -0,0 +1,43 @@
+    # Get hardware
+    hardware = get_hardware()
+    
+    print(f"Loading model from {local_dir} on {hardware}...")
```
Hạn chế dùng print, nên dùng logging nhiều hơn nhé sếp

quan-dang commented on assets/architecture.png:
Hình này có một vài vấn đề:

1. Nên đánh số cho mũi tên, và giải thích kỹ hơn các mũi tên có ý nghĩa gì 
2. Luồng CI/CD build&test không build vector DB, nên trỏ đúng vào nó build cái gì
3. User với file PDF sao lại cho vào 1 hình chữ nhập nhỉ sếp =)), nên là mũi tên upload đi từ user, và file pdf ở bên trên mũi tên. User query vector db thì mũi tên trỏ thẳng vào vector db chứ k trỏ vào hình chữ nhật bên ngoài.
4. Jaeger Tracing đâu sếp nhỉ?
5. Cái mũi tên chỗ `Push` lên Docker Hub cũng k rõ push gì lên đó, rồi ở bước pull thì cũng k biết pull image nào, pod nào pull, rồi trong k8s cluster có gì, cũng k biết External IP là của cái nào

Sếp tham khảo đồ án của một bạn khác ở đây coi: https://github.com/bmd1905/ChatOpsLLM, hoặc các đồ án khác ở: https://fullstackdatascience.com/hall-of-fame

quan-dang commented on README.md:
Chưa thể hiện code coverage nhé sếp.

hieunq95 committed:
feat: adding features and revise code

Adding Jager Tracing

Adding unified .env file for all services

Updating image tags for services in docker-compose build

Adding full coverage tests

hieunq95 committed:
fix: updating code-cov report

hieunq95 committed:
fix: updating Jenkins pipeline test & build

hieunq95 committed:
fix: updating codecov_token

Fixing test stage

hieunq95 committed:
fix: fixing typo

hieunq95 committed:
fix: installing python3.10-venv in Dockerfile

hieunq95 committed:
fix: updating Jenkinsfiles & Dockerfile

hieunq95 committed:
fix: removing sudo 666 permission on Jenkins

hieunq95 committed:
fix: fixing Jenkins CI/CD pipeline logic

hieunq95 committed:
chore(jenkins): update container build target

hieunq95 committed:
fix: fixing CI/CD pipeline

hieunq95 committed:
fix: fixing CI/CD pipeline

hieunq95 committed:
fix: updating CI/CD pipeline

hieunq95 committed:
fix: fixing CI/CD pipeline

hieunq95 committed:
fix: fixing CI/CD pipeline

hieunq95 committed:
fix: fixing CI/CD pipeline

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing check coverage Jenkins

hieunq95 committed:
fix: fixing Jenkins pipeline - upload coverage

hieunq95 committed:
fix: fixing Jenkins pipeline - upload coverage

hieunq95 committed:
fix: fixing Jenkins pipeline - upload coverage

hieunq95 committed:
fix: fixing Jenkins pipeline - upload coverage

hieunq95 committed:
fix: fixing Jenkins pipeline - upload coverage

hieunq95 committed:
fix: fixing Jenkins pipeline - upload coverage

hieunq95 committed:
fix: fixing Jenkins pipeline - upload coverage

hieunq95 committed:
fix: fixing Jenkins pipeline - upload coverage

hieunq95 committed:
fix: fixing Jenkins pipeline - upload coverage

hieunq95 committed:
fix: fixing Jenkins pipeline - upload coverage

hieunq95 committed:
fix: fixing CI/CD pipeline

hieunq95 committed:
fix: fixing CI/CD pipeline

hieunq95 committed:
fix: fixing CI/CD pipeline

hieunq95 committed:
fix: fixing CI/CD pipeline

hieunq95 committed:
fix: fixing CI/CD pipeline

hieunq95 committed:
fix: fixing code coverage in CI/CD pipeline

hieunq95 committed:
fix: fixing code coverage in CI/CD pipeline

hieunq95 committed:
feat: finalize CI/CD pipeline with code coverage check

hieunq95 committed:
fix: updating manual trace - Jaeger Tracing

fixing OpenBLAS Warning

hieunq95 committed:
fix: fixing pytest errors due to updates in src/

hieunq95 committed:
chore: update branch name Jenkinsfile

hieunq95 committed:
feat: add NGINX API gateway

hieunq95 committed:
fix: fixing resource limit - OOMKilled on kubernetes

updating resource allocations in kubernetes deployment

fixing nginx configuration

hieunq95 committed:
fix: volume path error

hieunq95 committed:
chore: update README and rename directory

hieunq95 commented:
Resolve all issues. Merge PR!
